mongo/jobs.go

```diff
@@ -122,52 +122,67 @@ func (m *JobStore) GetJob(ctx context.Context, id string) (*models.Job, error) {
 }
 
 // GetJobs retrieves all the jobs, from the collection, and lists them in order of last_updated
-func (m *JobStore) GetJobs(ctx context.Context, option Options) (models.Jobs, error) {
+func (m *JobStore) GetJobs(ctx context.Context, option Options) (*models.Jobs, error) {
 	logData := log.Data{"option": option}
 	log.Info(ctx, "getting list of jobs", logData)
 
-	s := m.Session.Copy()
-	defer s.Close()
-
-	results := models.Jobs{}
-
-	numJobs, err := s.DB(m.Database).C(m.JobsCollection).Count()
+	// get jobs count
+	numJobs, err := m.getJobsCount(ctx)
 	if err != nil {
-		log.Error(ctx, "error counting jobs", err, logData)
-		return results, err
+		log.Error(ctx, "failed to get jobs count", err, logData)
+		return nil, err
 	}
 
-	logData["no_of_jobs"] = numJobs
-	log.Info(ctx, "number of jobs found in jobs collection", logData)
-
-	if numJobs == 0 {
-		log.Info(ctx, "there are no jobs in the data store - so the list is empty", logData)
-		results.JobList = make([]models.Job, 0)
-		return results, nil
-	}
+	s := m.Session.Copy()
+	defer s.Close()
 
+	// get all jobs from mongo
 	jobsQuery := s.DB(m.Database).C(m.JobsCollection).Find(bson.M{}).Skip(option.Offset).Limit(option.Limit).Sort("last_updated")
 
+	// populate jobsList from jobsQuery
```

Comment:
(suggestion): again not sure if you need comments here.

Reply:
Same as **COMMENT 1**